src/Umbraco.Abstractions/Configuration/Configs.cs

```diff
@@ -15,8 +13,26 @@ namespace Umbraco.Core.Configuration
     /// </remarks>
     public class Configs
     {
+        private readonly Func<string, object> _configGetter;
+
+        public IFactory Factory
```

Comment:
I have refactored how this works since this is less than ideal. Now the call to `RegisterWith` includes a callback to acquire the factory which is stored as a lazy accessor.